src/core/plugins/spec/actions.js

```diff
@@ -53,8 +53,38 @@ export function updateUrl(url) {
   return {type: UPDATE_URL, payload: url}
 }
 
+// Build parent-child schema map
+function getParentToChildMap(json){
+  const schemaMap = new Map()
+  
+  if (json.components && json.components.schemas) {
+    Object.entries(json.components.schemas).forEach(([schemaName, schema]) => {
+      if (schema.allOf) {
+        schema.allOf.forEach(item => {
+          if (item.$ref) {
+            // Extract parent schema name from $ref
+            const parentName = item.$ref.split('/').pop()
+            // Add current schema as child of parent
+            if (!schemaMap.has(parentName)) {
+              schemaMap.set(parentName, [])
+            }
+            schemaMap.get(parentName).push(schemaName)
+          }
+        })
+      }
+    })
+  }
+  return schemaMap;
+}
+
 export function updateJsonSpec(json) {
-  return {type: UPDATE_JSON, payload: json}
+  return {
+    type: UPDATE_JSON, 
+    payload: {
+      json,
+      parentToChildMap: Object.fromEntries(getParentToChildMap(json))
```

Comment:
Is there a better spot / way to do this? We need the whole spec to produce the parent to child map, which should be a cheap operation because it just iterating through components once and storing names